非同期通信

Keitamidehyah opened the pull request:
WHAT
API の作成

WHY
必要なデータをjson形式で返すため

Keitamidehyah committed:
非同期通信

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,40 @@
+    e.preventDefault();
+    var formData = new FormData(this);
+    var url = $(this).attr('action')
+    console.log(url)
```
デバッグ消しましょう

TOMARUMARU commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,40 @@
+                    <a href=/users/${meessage.user_id}>${comment.user_name}</a>
+                    :
+                  </strong>
+                  ${content.meessage}
```
画像も送れるようにしましょう

Keitamidehyah committed:
非同期通信　修正

WHAT
指摘していただいたところとバグの修正

i-date commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,40 @@
+                    :
+                  </strong>
+                  ${comment.content}
+                </p>`
```
> 画像も送れるようにしましょう

上記コメントの修正がされていないようです。
`var html = 略`の前に（三項演算子を使用して条件によって値を変えた）変数を設け、その変数をバッククオート内で呼び出すようにするといいでしょう。その際、create.json.jbuilderでimageを渡すコードも追加する必要もあります。
```
var （message.imageの）変数名 = （条件） ? （trueの場合の値） : （falseの場合の値）;   // 三項演算子で値（imgタグor空）を決定している
var html = `略
  ${ 変数名 } で変数呼び出し
  略`
```

i-date commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,40 @@
+      contentType: false
+    })
+    .done(function(data){
+      console.log(data)
```
console.logは削除しましょう。

i-date commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,40 @@
+      console.log(data)
+      var html = buildHTML(data);
+      $('.comments').append(html);
+      $('.messagebox').val('');
```
31行目の`$('form').get(0).reset();`でリセットされるため、29行目の`$('.messagebox').val('');`は不要ですので削除しましょう。

i-date reviewed (commented):
追加コメントいたしました^^

Keitamidehyah committed:
非同期通信　修正２

WHAT
演算三項子を使用してimageを反映させるコード、console logの消去

ky3090 commented on app/controllers/messages_controller.rb:
```diff
@@ -9,12 +9,16 @@ def index
-      redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'
+      redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'  and return
+      respond_to do |format|
+      format.html { redirect_to group_messages_path, notice: "メッセージを送信しました" }
```
12行目は不要なので削除しましょう！
また、14行目もこのままだとエラー出てしまいます。

ky3090 commented on app/views/messages/create.json.jbuilder:
```diff
@@ -0,0 +1,4 @@
+json.content @message.content
+json.user_id   @message.user_id
+json.user_name   @message.user.name
+json.image         @message.image
```
> @message.image

この記述のままで、画像はしっかり表示されるでしょうか？

ky3090 commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,41 @@
+                    :
+                  </strong>
+                  ${comment.content}
+                </p>`
```
6 ~ 12行目の記述は、htmlは現存してらっしゃるでしょうか？
しっかり、現存する記述をこの部分に加えましょう。

Keitamidehyah committed:
非同期通信　修正３

WHAT
前回ご指摘いただいた部分の修正

WHY
message.jsを正常に動かすためのjbuilderとmessage.html.hamlの一致した記述

画像を投稿した際の反映

不要な部分の削除

miyagit reviewed (approved):
LGTM
